capstone/src/main/webapp/youtubeScript.js

```diff
@@ -347,3 +359,46 @@ async function getKeywordSearchResults() {
   });
   document.getElementById('search-type').innerHTML = "Keyword Search";
 }
+
+/** Prepares CSV download*/
+function prepareDownload(sheetHeader, sheetData, sheetName) {
+  let csv = sheetHeader.join(',') + '\n';
+  for (const data of sheetData) {
+    csv += data.join(',') + '\n';
```

Comment:
You might have problems here if the comment text contains any `,` or `\n`.  I think there isn't a 100% agreed upon standard for how to handle this, the pandas library we often use puts strings in "" if they contain either `,` or `\n` in the string?

So for example if you have 2 columns TEXT and TOXICITY and the string is "hello world" you would just write:
```
TEXT,TOXICITY
hello world,0.123
```
but if the text is `hello, world` the output would look like
```
TEXT,TOXICITY
"hello, world",0.123
```
and if it was `hello\nworld` then your single comment would actually span 2 lines like:

```
TEXT,TOXICITY
"hello
world",0.123
```
(where there is a `\n` between hello and world, it just would appear as a newline to the user)

Reply:
In my function "formatCommentForSpreadsheet" I attempt to remedy this problem. My current implementation accounts for excessive white space, commas and newlines by replacing them. Is there an implementation that would permit me to preserve comments that contain commas and allow them to be apart of the CSV without modification?

Reply:
This looks fine for now as it ensure the CSV files can be loaded.  If you want to preserve `,` and `\n` you might be able to do something similar to this, which wraps the comment in `"` if it contains any of those characters:
```
// Note, indexOf returns -1 if what you are searching for isn't found
if (comment.indexOf(',') >= 0 || comment.indexOf('\n') >= 0) {
   comment = '"' + comment + '"';
}
```